buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -224,22 +224,30 @@ synchronized void stop(boolean tailLogs) {
         }
         logger.info("Stopping `{}`, tailLogs: {}", this, tailLogs);
         requireNonNull(esProcess, "Can't stop `" + this + "` as it was not started or already stopped.");
-        stopHandle(esProcess.toHandle());
+        // Test clusters are not reused, don't spend time on a graceful shutdown
+        stopHandle(esProcess.toHandle(), true);
         if (tailLogs) {
             logFileContents("Standard output of node", getStdoutFile());
             logFileContents("Standard error of node", getStdErrFile());
         }
         esProcess = null;
     }
 
-    private void stopHandle(ProcessHandle processHandle) {
+    private void stopHandle(ProcessHandle processHandle, boolean forcibly) {
         // Stop all children first, ES could actually be a child when there's some wrapper process like on Windows.
         if (processHandle.isAlive()) {
-            processHandle.children().forEach(this::stopHandle);
+            processHandle.children().forEach(each -> stopHandle(each, forcibly));
         }
-        logProcessInfo("Terminating elasticsearch process:", processHandle.info());
+        logProcessInfo(
+            "Terminating elasticsearch process" + (forcibly ? " forcibly " : "gratefully") + ":",
+            processHandle.info()
+        );
         if (processHandle.isAlive()) {
-            processHandle.destroy();
+            if (forcibly) {
```

Comment:
I think the order of this all can be cleaned up to not have doubling up on waits.

Reply:
@rjernst done, ready for review

Reply:
We're still doing multiple aliveness checks when we already know it is killed. Here is what I was envisioning:

```
if (processHandle.isAlive() == false) {
    logger.info("Process was not running when we tried to terminate it.");
	return;
}

// Stop all children first, ES could actually be a child when there's some wrapper process like on Windows.
processHandle.children().forEach(each -> stopHandle(each, forcibly));

logProcessInfo("Terminating elasticsearch process" + (forcibly ? " forcibly " : "gratefully") + ":", processHandle.info());

if (forcibly) {
    processHandle.destroyForcibly();
} else {
    processHandle.destroy();
    waitForProcessToExit(processHandle);
    if (processHandle.isAlive()) {
	    logger.info("process did not terminate after {} {}, stopping it forcefully",
	                       ES_DESTROY_TIMEOUT, ES_DESTROY_TIMEOUT_UNIT);
	    processHandle.destroyForcibly();
    }
}

waitForProcessToExit(processHandle);
if (processHandle.isAlive()) {
    throw new TestClustersException("Was not able to terminate elasticsearch process");
}
```

Reply:
@rjernst implemented as suggested